### PR TITLE
Reject expired IDs

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Reject expired IDs](https://trello.com/c/TXOi8uy4/565-reject-expired-ids)

File: lib/brain.js [hunk 1]
```diff
@@ -676,11 +676,29 @@ Brain.prototype.scanPDF = function scanPDF () {
       return
     }
 
+    if (this.hasExpired(result)) {
+      console.log('Expired ID card')
```
Comment (josepfo) [inline lib/brain.js:680]: If this console.log isn't needed remove it and move this to Rafa's

**Review by josepfo — CHANGES_REQUESTED** [review]

**Review by josepfo — APPROVED** [review]

File: lib/brain.js [hunk 2]
```diff
@@ -676,11 +676,29 @@ Brain.prototype.scanPDF = function scanPDF () {
...
+Brain.prototype.hasExpired = function hasExpired (cardData) {
+  if (cardData.country === 'ZA') return false
+  const expirationDay = cardData.expirationDay
+  const expirationMonth = cardData.expirationMonth
+  const expirationYear = cardData.expirationYear
+  const expirationDate = expirationDay
+    ? new Date(expirationYear, expirationMonth, expirationDay)
+    : new Date(expirationYear, expirationMonth)
+
+  const now = Date.now()
```
Comment (RafaelTaranto) [inline lib/brain.js:698]: Isn't this a issue for cards that are expiring on the month of the tx?

File: lib/compliance/parsepdf417-us.js [hunk 3]
```diff
@@ -97,7 +97,14 @@ function normalize (result, version, issuingAuthorityNumber) {
 
   if (result.country === 'USA' && version > 1) {
     result.dateOfBirth = date.substr(4, 4) + date.substr(0, 4)
+    result.expirationDay = expirationDate ? expirationDate.substr(2, 2) : '00'
+    result.expirationMonth = expirationDate ? expirationDate.substr(0, 2) : '00'
+    result.expirationYear = expirationDate ? expirationDate.substr(4, 4) : '00'
```
Comment (RafaelTaranto) [inline lib/compliance/parsepdf417-us.js:102]: Can't we just use the current expirationDate to perform the test? We don't really need to create the additional fields.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Review by RafaelTaranto — APPROVED** [review]

File: lib/brain.js [hunk 4]
```diff
@@ -676,11 +676,27 @@ Brain.prototype.scanPDF = function scanPDF () {
...
+      return idCardData.dispatch('SCAN_ERROR')
+    }
+
     idCardData.setData(result)
     return idCardData.dispatch('SCANNED')
   })
 }
 
+Brain.prototype.hasExpired = function hasExpired (cardData) {
+  if (cardData.country === 'ZA') return false
```
Comment (joshmh) [inline lib/brain.js:690]: Explain why we're returning false here.

**Review by joshmh — CHANGES_REQUESTED** [review]

**Pull request merged**